src/OpenApi/src/Services/Schemas/OpenApiSchemaService.cs

```diff
@@ -213,13 +213,117 @@ private async Task InnerApplySchemaTransformersAsync(OpenApiSchema schema,
             }
         }
 
-        if (schema is { AdditionalPropertiesAllowed: true, AdditionalProperties: not null } && jsonTypeInfo.ElementType is not null) 
-		{
+        if (schema is { AdditionalPropertiesAllowed: true, AdditionalProperties: not null } && jsonTypeInfo.ElementType is not null)
+        {
             var elementTypeInfo = _jsonSerializerOptions.GetTypeInfo(jsonTypeInfo.ElementType);
             await InnerApplySchemaTransformersAsync(schema.AdditionalProperties, elementTypeInfo, null, context, transformer, cancellationToken);
         }
-	}
+    }
 
     private JsonNode CreateSchema(OpenApiSchemaKey key)
-        => JsonSchemaExporter.GetJsonSchemaAsNode(_jsonSerializerOptions, key.Type, _configuration);
+    {
+        var sourceSchema = JsonSchemaExporter.GetJsonSchemaAsNode(_jsonSerializerOptions, key.Type, _configuration);
+
+        // Resolve any relative references in the schema
+        ResolveRelativeReferences(sourceSchema, sourceSchema);
+
+        return sourceSchema;
+    }
+
+    // Helper method to recursively resolve relative references in a schema
+    private static void ResolveRelativeReferences(JsonNode node, JsonNode rootNode)
+    {
+        if (node is JsonObject jsonObj)
+        {
+            // Check if this node has a $ref property with a relative reference and no schemaId to
+            // resolve to
+            if (jsonObj.TryGetPropertyValue(OpenApiSchemaKeywords.RefKeyword, out var refNode) &&
```

Comment:
Not having a schema ID indicates that the relative reference is to something that isn't a complex type, for example a `List<string>`. For things that have a schema ID, we create a ref to the schema in the `OpenApiComponents` via the `OpenApiSchemaReferenceTransformer`.